Sources/PrimerSDK/Classes/User Interface/TokenizationViewModels/IPay88TokenizationViewModel.swift

```diff
@@ -0,0 +1,547 @@
+        let numberFormatter = NumberFormatter()
+        numberFormatter.numberStyle = .decimal
+        numberFormatter.decimalSeparator = "."
+        numberFormatter.groupingSeparator = ","
+        numberFormatter.maximumFractionDigits = 2
+        numberFormatter.minimumFractionDigits = 2
+        return numberFormatter
+    }()
+    
+    override func validate() throws {
+        guard let decodedJWTToken = PrimerAPIConfigurationModule.decodedJWTToken, decodedJWTToken.isValid else {
+            let err = PrimerError.invalidClientToken(
+                userInfo: ["file": #file, "class": "\(Self.self)", "function": #function, "line": "\(#line)"],
+                diagnosticsId: nil)
+            ErrorHandler.handle(error: err)
+            throw err
+        }
+        
+        guard decodedJWTToken.pciUrl != nil else {
+            let err = PrimerError.invalidClientToken(userInfo: ["file": #file, "class": "\(Self.self)", "function": #function, "line": "\(#line)"], diagnosticsId: nil)
+            ErrorHandler.handle(error: err)
+            throw err
+        }
+        
+        var errors: [PrimerError] = []
+        
+        // Merchant info
+        
+        if self.config.id == nil {
+            let err = PrimerError.invalidValue(
+                key: "configuration.id",
+                value: config.id,
+                userInfo: ["file": #file, "class": "\(Self.self)", "function": #function, "line": "\(#line)"],
+                diagnosticsId: nil)
+            ErrorHandler.handle(error: err)
+            errors.append(err)
+        }
+        
+        if (self.config.options as? MerchantOptions)?.merchantId == nil {
+            let err = PrimerError.invalidValue(
+                key: "configuration.merchantId",
+                value: config.id,
+                userInfo: ["file": #file, "class": "\(Self.self)", "function": #function, "line": "\(#line)"],
+                diagnosticsId: nil)
+            ErrorHandler.handle(error: err)
+            errors.append(err)
+        }
+        
+        // Amount & currency validation
+        
+        if AppState.current.amount == nil {
+            let err = PrimerError.invalidClientSessionValue(
+                name: "amount",
+                value: nil,
+                userInfo: ["file": #file, "class": "\(Self.self)", "function": #function, "line": "\(#line)"],
+                diagnosticsId: nil)
+            ErrorHandler.handle(error: err)
+            errors.append(err)
+        }
+        
+        if AppState.current.currency != .MYR {
+            let err = PrimerError.invalidClientSessionValue(
+                name: "currency",
+                value: AppState.current.currency?.rawValue,
+                userInfo: ["file": #file, "class": "\(Self.self)", "function": #function, "line": "\(#line)"],
+                diagnosticsId: nil)
+            ErrorHandler.handle(error: err)
+            errors.append(err)
+        }
```

Comment:
These checks should be done before showing the payment method button to the user / passing it to `onAvailablePaymentMethodsLoad`.

No need to show a payment method if know it won't work

Reply:
Isn't this supposed to be driven by the BE? If a PM is not valid for a PM, then BE shouldn't return it in the response, and therefore not rendered. Right?

Reply:
@EvansPie similar to what we did to other APMs. We verify it **before** and log warnings if something is wrong. Probably you did it for Apple Pay